examples/javaGraphics/gfx.fz

```diff
@@ -100,12 +100,12 @@ is
   sin240 := -866
 
 
-  G2D (g Java.java.awt.Graphics, tx, ty, rot0 i32) ref is  # NYI: #167: 'ref' due to lack of 'like this'
+  G2D (g Java.java.awt.Graphics, tx, ty, rot0 i32) is
     rot := if rot0 >= 0 rot0 % 360 else rot0 % 360 + 360
 
     rotate (d i32) => G2D g tx ty rot+d
 
-    threeTimes(d (G2D, Java.java.awt.Color) -> unit) unit is
+    threeTimes(d (ref G2D, Java.java.awt.Color) -> unit) unit is
```

Comment:
I would like to have this: 
```
 fixed rotate (d i32) G2D.this.type is G2D g tx ty rot+d
 threeTimes(d (G2D.this.type, Java.java.awt.Color) -> unit) unit is
```
But `fixed` does not work like this yet. I will create an issue for that.

Reply:
Trying to add an issue, I realized this `fixed` actuall works already, see this example: 
```
test_fixed is

  A is
    fixed me A.this.type is A.this
    redef as_string => "A"

  B : A is
    fixed me => B.this
    redef as_string => "B"

  say A.me
  say B.me
```
that produces
```
> ./build/bin/fz test_fixed2.fz 
A
B
```
Can you try to change `rotate` to be `fixed` with result type `G2D.this.type` and change the first argument type of `threeTimes.d` to `G2d.this.type`?

Reply:
I played aroudn with this a bit more to make sure that `B.me` is actually called, and it is:

```
test_fixed is

  A is
    fixed me A.this.type is A.this
    mer ref A is me
    redef as_string => "A"

  B(s String) : A is
    fixed me => (B "new")
    redef as_string => "B$s"

  say A.me
  say (B "orig").me

  boxed(r ref A) => say r.mer
  boxed A
  boxed (B "orig")
```
results in
```
 > ./build/bin/fz test_fixed2.fz 
A
Bnew
A
Bnew
```
Note that a call to `r.me` within `boxed` results in an error due to ambiguous result types.

Reply:
I was wrong, `fixed` is not powerful enough for this yet, the following example fails

```
test_fixed is

  A is
    fixed me A.this.type is A
    redef as_string => "A"

  B(s String) : A is
    fixed me => (B "new")
    redef as_string => "B$s"

  say A.me
  say (B "orig").me
```
as follows
```
 > ./build/bin/fz test_fixed4.fz 

/home/fridi/fuzion/clean/fuzion/test_fixed4.fz:4:29: error 1: Incompatible types in assignment
    fixed me A.this.type is A
----------------------------^
assignment to field : 'test_fixed.A.me.result'
expected formal type: 'test_fixed.this.type.A.this.type'
actual type found   : 'test_fixed.this.type.A'
assignable to       : 'test_fixed.this.type.A'
for value assigned  : 
{
  box(deref(deref(me.this.#^test_fixed.A.me).#^test_fixed.A).A)
}

To solve this, you could change the type of the target 'test_fixed.A.me.result' to 'test_fixed.this.type.A' or convert the type of the assigned value to 'test_fixed.this.type.A.this.type'.

one error.
```